js/settingsHelpers.js

```diff
@@ -94,15 +94,21 @@ export const hideOptionalElementsOnShowForm = elementsArray => {
   });
 };
 
-export const handleContactInformationRadioButtonPresets = (canWeVoicemailMobile, canWeText, canWeVoicemailHome, canWeVoicemailOther) => {
-  document.getElementById('mobileVoicemailPermissionYesRadio').checked = canWeVoicemailMobile;
-  document.getElementById('mobileVoicemailPermissionNoRadio').checked = !canWeVoicemailMobile;
-  document.getElementById('textPermissionYesRadio').checked = canWeText;
-  document.getElementById('textPermissionNoRadio').checked = !canWeText;
-  document.getElementById('homeVoicemailPermissionYesRadio').checked = canWeVoicemailHome;
-  document.getElementById('homeVoicemailPermissionNoRadio').checked = !canWeVoicemailHome;
-  document.getElementById('otherVoicemailPermissionYesRadio').checked = canWeVoicemailOther;
-  document.getElementById('otherVoicemailPermissionNoRadio').checked = !canWeVoicemailOther;
+export const handleContactInformationRadioButtonPresets = (mobilePhoneNumberComplete, canWeVoicemailMobile, canWeText, homePhoneNumberComplete, canWeVoicemailHome, otherPhoneNumberComplete, canWeVoicemailOther) => {
+  if (mobilePhoneNumberComplete) {
+    document.getElementById('mobileVoicemailPermissionYesRadio').checked = canWeVoicemailMobile;
+    document.getElementById('mobileVoicemailPermissionNoRadio').checked = !canWeVoicemailMobile;
+    document.getElementById('textPermissionYesRadio').checked = canWeText;
```

Comment:
Do HTML radio buttons not automatically un-select if another radio button in the same group is selected?

Reply:
They sure do. This method pre-loads the previously selected values based on user profile data when a user opens the form to edit.